interfaces/MatlabToolbox/matlab_new/Thermo/ThermoPhase.m

```diff
@@ -0,0 +1,1736 @@
+        end
+
+        function set.STX(tp, input)
+            tp.X = input{3};
+            tp.ST = input(1:2);
+        end           
+
+        function set.STY(tp, input)
+            tp.Y = input{3};
+            tp.ST = input(1:2);
+        end
+         
+        function set.SV(tp, input)
+            s = input{1};
+            v = input{2};
+            if v <= 0
+                error('The specific volume must be positive');
+            end
+            calllib(ct, 'thermo_set_SV', tp.tpID, [s, v]);            
+        end
+
+        function set.SVX(tp, input)
+            tp.X = input{3};
+            tp.SV = input(1:2);
+        end           
+
+        function set.SVY(tp, input)
+            tp.Y = input{3};
+            tp.SP = input(1:2);
+        end            
+        
+        function set.TD(tp, input)
+            t = input{1};
+            d = input{2};
+            if t <= 0
+                error('The temperature must be positive');
+            end
+            if d <= 0
+                error('The density must be positive');
+            end
+            tp.T = t;
+            tp.D = d;         
+        end        
+        
+        function set.TDX(tp, input)
+            tp.X = input{3};              
+            tp.TD = input(1:2);      
+        end        
+        
+        function set.TDY(tp, input)
+            tp.Y = input{3};             
+            tp.TD = input(1:2);        
+        end
+        
+        function set.TH(tp, input)
+            t = input{1};
+            if t <= 0
+                error('The temperature must be positive');
+            end
+            h = input{2};
+            calllib(ct, 'thermo_set_TH', tp.tpID, [t, h]);            
+        end
+
+        function set.THX(tp, input)
+            tp.X = input{3};
+            tp.TH = input(1:2);
+        end           
+
+        function set.THY(tp, input)
+            tp.Y = input{3};
+            tp.TH = input(1:2);
+        end           
+        
+        function set.TP(tp, input)
+            t = input{1};
+            p = input{2};
+            if t <= 0
+                error('The temperature must be positive');
+            end
+            if p <= 0
+                error('The pressure must be positive');
+            end
+            tp.T = t;
+            tp.P = p;
+        end        
+                   
+        function set.TPX(tp, input)
+            tp.X = input{3};
+            tp.TP = input(1:2);
+        end
+       
+        function set.TPY(tp, input)
+            tp.Y = input{3};            
+            tp.TP = input(1:2);
+        end        
+        
+        function set.TV(tp, input)
+            t = input{1};
+            v = input{2};
+            if t <= 0
+                error('The temperature must be positive');
+            end
+            if v <= 0
+                error('The specific volume must be positive');
+            end
```

Comment:
This error checking is all redundant -- the underlying C++ functions also check that the input temperature / pressure / density values are positive.